src/Raven.Server/Documents/ETL/EtlLoader.cs

```diff
@@ -157,14 +158,32 @@ public virtual void Dispose()
         {
             _database.Changes.OnDocumentChange -= NotifyAboutWork;
 
-            Parallel.ForEach(_processes, x => x.Dispose());
+            var ea = new ExceptionAggregator(Logger, "Could not dispose ETL loader");
```

Comment:
Do you mean `ExceptionAggregator`? It uses `ConcurrentSet` under the hood so it should be fine